cuda_core/cuda/core/experimental/_memory.py

```diff
@@ -326,6 +326,28 @@ def __init__(self, device_id: int):
         self._handle = handle_return(driver.cuDeviceGetMemPool(device_id))
         self._dev_id = device_id
 
+        # Set a higher release threshold to improve performance when there are no active allocations.
+        # By default, the release threshold is 0, which means memory is immediately released back
+        # to the OS when there are no active suballocations, causing performance issues.
+        try:
+            # Check current release threshold
+            current_threshold = handle_return(
+                driver.cuMemPoolGetAttribute(self._handle, driver.CUmemPool_attribute.CU_MEMPOOL_ATTR_RELEASE_THRESHOLD)
+            )
+            # If threshold is 0 (default), set it to maximum to retain memory in the pool
+            if int(current_threshold) == 0:
+                handle_return(
+                    driver.cuMemPoolSetAttribute(
+                        self._handle,
+                        driver.CUmemPool_attribute.CU_MEMPOOL_ATTR_RELEASE_THRESHOLD,
+                        driver.cuuint64_t(0xFFFFFFFFFFFFFFFF),
+                    )
+                )
+        except Exception:
+            # If setting the attribute fails (e.g., on older drivers), continue without error
+            # to maintain backward compatibility
+            pass
```

Comment:
We don't need this try-except. This API will always work within our support matrix, and I want the exception, if any, to be raised to the user.